test/entrypoint/srv_ddl_reload.lua

```diff
@@ -0,0 +1,198 @@
+                    box.space['_ddl_sharding_func']:truncate()
+                end
+
+                local _, err = ddl.set_schema(schema)
+                if err ~= nil then
+                    error(err)
+                end
+            end)
+
+            rawset(_G, 'set_sharding_key', function(space_name, sharding_key_def)
+                if box.info.ro == true then
+                    return
+                end
+
+                local current_schema, err = ddl.get_schema()
+                if err ~= nil then
+                    error(err)
+                end
+
+                box.space['_ddl_sharding_key']:replace{space_name, sharding_key_def}
+                current_schema.spaces[space_name].sharding_key = sharding_key_def
+
+                local _, err = ddl.set_schema(current_schema)
+                if err ~= nil then
+                    error(err)
+                end
+            end)
+
+            rawset(_G, 'set_sharding_func_name', function(space_name, sharding_func_name)
+                if box.info.ro == true then
+                    return
+                end
+
+                local current_schema, err = ddl.get_schema()
+                if err ~= nil then
+                    error(err)
+                end
+
+                local t = {space_name, sharding_func_name, box.NULL}
+                box.space['_ddl_sharding_func']:replace(t)
+                current_schema.spaces[space_name].sharding_func = sharding_func_name
+
+                local _, err = ddl.set_schema(current_schema)
+                if err ~= nil then
+                    error(err)
+                end
+            end)
+
+            rawset(_G, 'set_sharding_func_body', function(space_name, sharding_func_body)
+                if box.info.ro == true then
+                    return
+                end
+
+                local current_schema, err = ddl.get_schema()
+                if err ~= nil then
+                    error(err)
+                end
+
+                local t = {space_name, box.NULL, sharding_func_body}
+                box.space['_ddl_sharding_func']:replace(t)
+                current_schema.spaces[space_name].sharding_func = { body = sharding_func_body }
+
+                local _, err = ddl.set_schema(current_schema)
+                if err ~= nil then
+                    error(err)
+                end
+            end)
+
+            rawset(_G, 'create_new_space', function()
+                if box.info.ro == true then
+                    return
+                end
+
+                local new_schema = table.deepcopy(schema)
+                new_schema.spaces['customers_new'] = table.deepcopy(customers_schema)
+                new_schema.spaces['customers_new'].sharding_func = {
+                    body = [[
+                        function(key)
+                            local vshard = require('vshard')
+                            return vshard.router.bucket_id_mpcrc32(key) % 3000 + 1
```

Comment:
Nit: It is a bit strange to apply `x % 3000 + 1` on `bucket_id`, which is already a hash by modulo of `bucket_count` plus one, where `bucket_count` is set below in `cartridge.cfg()` and is 3000.

Reply:
Fixed, thanks!